resources/views/portal/ninja2020/gateways/blockonomics/pay.blade.php

```diff
@@ -0,0 +1,321 @@
+
+            try {
+                const newPrice = await getBTCPrice();
+                if (newPrice) {
+                    // Update the text content of the countdown span to the new bitcoin price
+                    document.getElementsByClassName("btc-value")[0].innerHTML = "1 BTC = " + (newPrice || "N/A") + " {{$currency}}, updates in <span id='countdown'></span>";
+                    const newBtcAmount = ({{$amount}} / newPrice).toFixed(10);
+
+                    // set the value of the input field and the text content of the span to the new bitcoin amount
+                    document.querySelector('input[name="btc_price"]').value = newPrice;
+                    document.querySelector('input[name="btc_amount"]').value = newBtcAmount;
+                    document.getElementById('btc-amount').textContent = newBtcAmount;
+                    
+                    // set the href attribute of the link to the new bitcoin amount
+                    const qrCodeLink = document.getElementById('qr-code-link');
+                    const openInWalletLink = document.getElementById('open-in-wallet-link');
+                    qrCodeLink.href = `bitcoin:{{$btc_address}}?amount=${newBtcAmount}`;
+                    openInWalletLink.href = `bitcoin:{{$btc_address}}?amount=${newBtcAmount}`;
+                    
+                    // fetch and display the new QR code
+                    fetchAndDisplayQRCode(newBtcAmount);
+                    startTimer(600); // Restart timer for 10 minutes (600 seconds)
+                }
+            } finally {
+                refreshIcon.classList.remove('rotating');
+            }
+        }
+
+        const connectToWebsocket = () => {
+            const webSocketUrl = "wss://www.blockonomics.co/payment/{{ $btc_address }}";
+            const ws = new WebSocket(webSocketUrl);
+
+            ws.onmessage = function(event) {
+                const data = JSON.parse(event.data);
+                console.log('Payment status:', data.status);
+                const isPaymentUnconfirmed = data.status === 0;
+                const isPaymentPartiallyConfirmed = data.status === 1;
+                // TODO: Do we need to handle Payment confirmed status?
+                // This usually takes too long to happen, so we can just wait for the unconfirmed status?
+                if (isPaymentUnconfirmed || isPaymentPartiallyConfirmed) {
+                    document.querySelector('input[name="txid"]').value = data.txid || '';
+                    document.getElementById('server-response').submit();
+                }
+            }
+        };
+
+        const fetchAndDisplayQRCode = async (newBtcAmount = null) => {
+            try {
+                const btcAmount = newBtcAmount || '{{$btc_amount}}';
+                const response = await fetch(`/api/v1/get-blockonomics-qr-code?qr_string=bitcoin:${btcAmount}?amount={{$btc_amount}}`);
```

Comment:
This query string seems to have two quetion marks in the link. The second parameter(amount) should be preceded with an ampersand instead of a question mark

Reply:
Great catch, thanks!

Reply:
Awesome!